compiler/compile/OMRCompilation.cpp

```diff
@@ -698,16 +698,39 @@ bool OMR::Compilation::isPotentialOSRPointWithSupport(TR::TreeTop *tt)
 
    if (potentialOSRPoint && !self()->getOption(TR_FullSpeedDebug))
       {
-      // When in OSR HCR mode we need to make sure we check the BCI of the original
-      // call node to ensure we see the correct state of the doNotProfile flag
+
       if (self()->getHCRMode() == TR::osr &&
           (node->getOpCode().isCheck() || node->getOpCodeValue() == TR::treetop))
+         {
+         // When in OSR HCR mode we need to make sure we check the BCI of the original
+         // call node to ensure we see the correct state of the doNotProfile flag
```

Comment:
There is nothing explicit about calls here - I think this applies to anything with a ref count greater than 1, just at present only calls and asyncchecks are transition points and asyncchecks are not commoned - I don't know if the call needs to be handled differently so this comment may be spurious.